src/Translation/ItranslateApi.php

```diff
@@ -81,14 +85,37 @@ private function transformLanguageCode(?string $language): ?string
       return null;
     }
 
+    if (in_array($language, array_keys(self::LONG_LANGUAGE_CODE), true)) {
+      $long_codes = self::LONG_LANGUAGE_CODE[$language];
+      return $this->chooseWithoutPoliticalBias($long_codes);
+    }
+
     if (2 == strlen($language)) {
       return $language;
     }
 
-    if (in_array($language, self::LONG_LANGUAGE_CODE, true)) {
+    $long_codes = $this->array_flatten(array_values(self::LONG_LANGUAGE_CODE));
+    if (in_array($language, $long_codes, true)) {
       return $language;
     }
 
     return substr($language, 0, 2);
   }
+
+  private function array_flatten(array $array_of_arrays): array
+  {
+    $result = array();
+    foreach ($array_of_arrays as $array) {
+      foreach ($array as $element) {
+        $result[] = $element;
+      }
+    }
+
+    return $result;
+  }
+
+  private function chooseWithoutPoliticalBias(array $long_language_codes): string
```

Comment:
a bit of political humor here. If you think it has no place here, say so.

Reply:
I think this is fine if you are all good with it 👍